libraries/AR_WPNav/AR_WPNav.cpp

```diff
@@ -315,10 +323,17 @@ void AR_WPNav::update_pivot_active_flag()
         return;
     }
 
-    // if within 10 degrees of the target heading, exit pivot steering
-    if (yaw_error < AR_WPNAV_PIVOT_ANGLE_ACCURACY) {
+    uint32_t now = AP_HAL::millis();
+
+    // if within 10 degrees of the target heading, set start time of pivot steering
+    if (yaw_error < AR_WPNAV_PIVOT_ANGLE_ACCURACY && !_pivot_start_ms) {
+        _pivot_start_ms = now;
+    }
+
+    // exit pivot steering after the time set by pivot_delay has elapsed
+    if (_pivot_start_ms && now - _pivot_start_ms >= _pivot_delay * 1000) {
```

Comment:
Let's put a MAX(_pivot_delay, 0) to ensure that negative flow values are treated the same as zero.

Also have you checked that floating point times work?

Reply:
thank you for the review!
I fixed to work with floating value.